File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/Preferences.java
/**
 * 
 * @file
 *
 * @brief
 *
 * @author vitamin.caig@gmail.com
 * 
 */

package app.zxtune;

import android.content.Context;
import android.content.SharedPreferences;

public class Preferences {

  private Preferences() {
  }

  public static SharedPreferences getDefaultSharedPreferences(Context context) {
    return context.getSharedPreferences(getDefaultSharedPreferencesName(context),
          getDefaultSharedPreferencesMode());
  }

  private static String getDefaultSharedPreferencesName(Context context) {
    return context.getPackageName() + "_preferences";
  }
  
  private static int getDefaultSharedPreferencesMode() {
    return Context.MODE_PRIVATE | Context.MODE_MULTI_PROCESS;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/preferences/Preferences.kt
/**
 *
 * @file
 *
 * @brief
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.preferences

import android.content.Context
import android.content.SharedPreferences

object Preferences {

    private lateinit var providerClient: ProviderClient
    private lateinit var dataStore: DataStore

    @JvmStatic
    fun getDefaultSharedPreferences(context: Context): SharedPreferences =
        context.getSharedPreferences("${context.packageName}_preferences", Context.MODE_PRIVATE)

    fun getProviderClient(context: Context): ProviderClient {
        if (!this::providerClient.isInitialized) {
            providerClient = ProviderClient(context.applicationContext)
        }
        return providerClient
    }

    @JvmStatic
    fun getDataStore(context: Context): DataStore {
        if (!this::dataStore.isInitialized) {
            dataStore = DataStore(getProviderClient(context))
        }
        return dataStore
    }
}
